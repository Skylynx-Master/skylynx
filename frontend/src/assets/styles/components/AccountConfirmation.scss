@import "../Vars.scss";

.AccountConfirmation {
  width: 660px;
  height: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 8% auto 0;
  background-color: $color-white;
  padding: 20px;
  text-align: center;
  border-radius: 20px;
}

.AccountConfirmation h2 {
  font-size: 35px;
  padding: 50px 0;
  margin: 15px 0;
}

.AccountConfirmation img {
  width: 150px;
  height: 150px;
}

.AccountConfirmation button {
  width: 230px;
  padding: 10px;
  color: $color-typography;
  background-color: $color-peach;
  border-radius: 20px;
  font-size: 30px;
  margin: 25px 0;
  border: none;
  outline: none;
  cursor: pointer;
}
