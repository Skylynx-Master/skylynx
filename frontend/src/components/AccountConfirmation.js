import React from "react";
import "../assets/styles/components/AccountConfirmation.scss";
import checkIcon from "../assets/static/checkIcon.png";
import { Link } from "react-router-dom";

const AccountConfirmation = () => {
  return (
    <div className="AccountConfirmation">
      <h2 className="AccountConfirmation-title">
        Se ha confirmado su cuenta!!!
      </h2>
      <img
        src={checkIcon}
        alt="check image of account"
        className="AccountConfirmation-image"
      ></img>
      <Link to="/login">
        <button className="AccountConfirmation-button" type="button">
          Iniciar Sesi√≥n
        </button>
      </Link>
    </div>
  );
};

export default AccountConfirmation;
