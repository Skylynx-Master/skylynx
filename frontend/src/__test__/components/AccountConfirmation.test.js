import React from "react";
import AccountConfirmation from "../../components/AccountConfirmation";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";

describe("AccountConfirmation Component Testing", () => {
  const confirmation = mount(
    <BrowserRouter>
      <AccountConfirmation />
    </BrowserRouter>
  );

  test("Account Confirmation Render", () => {
    expect(confirmation.length).toEqual(1);
  });

  test("Account Confirmation title Render", () => {
    expect(confirmation.find(".AccountConfirmation-title").length).toEqual(1);
  });

  test("Account Confirmation title text", () => {
    expect(confirmation.find(".AccountConfirmation-title").text()).toEqual(
      "Se ha confirmado su cuenta!!!"
    );
  });

  test("Account Confirmation image render", () => {
    expect(confirmation.find(".AccountConfirmation-image").length).toEqual(1);
  });

  test("Account Confirmation button render", () => {
    expect(confirmation.find(".AccountConfirmation-button").length).toEqual(1);
  });
  test("Account Confirmation button text", () => {
    expect(confirmation.find(".AccountConfirmation-button").text()).toEqual(
      "Iniciar Sesi√≥n"
    );
  });
});
