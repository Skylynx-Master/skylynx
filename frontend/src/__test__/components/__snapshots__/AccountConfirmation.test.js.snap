// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountConfirmation Snapshot Test UI AccountConfirmation component 1`] = `
<div
  className="AccountConfirmation"
>
  <h2
    className="AccountConfirmation-title"
  >
    Se ha confirmado su cuenta!!!
  </h2>
  <img
    alt="check image of account"
    className="AccountConfirmation-image"
    src=""
  />
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      className="AccountConfirmation-button"
      type="button"
    >
      Iniciar Sesi√≥n
    </button>
  </a>
</div>
`;
